ssl/tls13_enc.c

```diff
@@ -507,7 +538,13 @@ int tls13_change_cipher_state(SSL_CONNECTION *s, int which)
             if (!ssl_cipher_get_evp_cipher(sctx, sslcipher, &cipher)) {
                 /* Error is already recorded */
                 SSLfatal_alert(s, SSL_AD_INTERNAL_ERROR);
-                EVP_MD_CTX_free(mdctx);
```

Comment:
I'd suggest that we move the creation of mdctx until just before the call to `ssl_md` below

Reply:
👍